packages/template-retail-react-app/app/components/_app/index.test.js

```diff
@@ -82,4 +85,68 @@ describe('App', () => {
         expect(hreflangLinks.some((link) => hasGeneralLocale(link))).toBe(true)
         expect(hreflangLinks.some((link) => link.hrefLang === 'x-default')).toBe(true)
     })
+
+    test('App component updates the basket with correct currency and customer email', async () => {
+        // Test basket. _app will be manipulating this basket for this test
+        const basket = {
+            basketId: 'basket_id',
+            currency: 'CAD',
+            customerInfo: {
+                customerId: 'customer_id'
+            }
+        }
+
+        jest.spyOn(
+            jest.requireActual('@salesforce/retail-react-app/app/hooks/use-current-customer'),
+            'useCurrentCustomer'
+        ).mockImplementation(() => {
+            return {
+                data: {
+                    customerId: 'customer_id',
+                    email: 'email@test.com',
+                    isRegistered: true
+                }
+            }
+        })
+
+        jest.spyOn(
+            jest.requireActual('@salesforce/retail-react-app/app/hooks/use-current-basket'),
+            'useCurrentBasket'
+        ).mockImplementation(() => {
+            return {
+                data: basket,
+                derivedData: {
+                    hasBasket: true,
+                    totalItems: 0
+                }
+            }
+        })
+
+        global.server.use(
+            // mock updating basket currency
+            rest.patch('*/baskets/:basketId', (req, res, ctx) => {
+                basket.currency = 'GBP'
+                return res(ctx.json(basket))
+            }),
+            // mock adding guest email to basket
+            rest.put('*/baskets/:basketId/customer', (req, res, ctx) => {
+                basket.customerInfo.email = 'customer@test.com'
+                return res(ctx.json(basket))
+            })
+        )
+
+        useMultiSite.mockImplementation(() => resultUseMultiSite)
+        await act(() =>
+            renderWithProviders(
+                <App
+                    targetLocale={DEFAULT_LOCALE}
+                    defaultLocale={DEFAULT_LOCALE}
+                    messages={messages}
+                />
+            )
+        )
+
+        expect(basket.currency).toBe('GBP')
+        expect(basket.customerInfo.email).toBe('customer@test.com')
```

Comment:
Why is basket expecting `customer@test.com`? The customer email is `email@test.com`, if customer.email !== basket.customerInfo.email, then the basket should get upated with customer.email. In this test, should it be `email@test.com` instead